dot/state/block.go

```diff
@@ -250,6 +251,23 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetHashesByNumber returns the block hashes with the given number
+func (bs *BlockState) GetHashesByNumber(num uint) ([]common.Hash, error) {
+	block, err := bs.GetBlockByNumber(num)
+	if err != nil {
+		return []common.Hash{}, fmt.Errorf("getting block by number %w", err)
```

Comment:
no point allocating an empty slice

```suggestion
		return nil, fmt.Errorf("getting block by number: %w", err)
```

(also missing `:` at the end)